pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroRecordExtractor.java

```diff
@@ -107,4 +121,25 @@ protected Object convertRecord(Object value) {
     }
     return convertedMap;
   }
+
+  /**
+   * This method convert any Avro logical-type converted (or not) value to a class supported by
+   * Pinot {@link GenericRow}
+   *
+   * Note that at the moment BigDecimal is converted to Pinot double which may lead to precision loss or may not be
+   * represented at all.
+   * Similarly, timestamp microsecond precision is not supported at the moment. These values will get converted to
+   * millisecond precision.
+   */
+  @Override
+  protected Object convertSingleValue(Object value) {
+    if (value instanceof BigDecimal) {
+      return ((BigDecimal) value).doubleValue();
+    }
+    if (value instanceof Instant) {
```

Comment:
I'm thinking it may be better to initially return a String representation of the timestamp instead, Pinot can coerce the value into a timestamp. At this point I can't make a guess of whether the user has specified millis or micros. Behaviour control could be configured via record reader settings in the future. @Jackie-Jiang  @walterddr let me know what you think

Reply:
Seems the `Instant` is always converted from the timestamp type, so we might want to return the `Timestamp` object here.

Reply:
this now returns `java.sql.Timestamp` instead